### ISSUE TITLE
UFCS bypasses feature checking

### ISSUE THREAD

**Issue by sfackler** [rust-lang/rust#22641]
```rust
use std::num::Int;

fn main() {
    let _x = <u32 as Int>::max_value();
    let _y: u32 = Int::max_value();
}
```
```
test.rs:5:19: 5:33 warning: use of unstable library feature 'core': unsure about its place in the world
test.rs:5     let _y: u32 = Int::max_value();
                            ^~~~~~~~~~~~~~
test.rs:5:19: 5:33 help: add #![feature(core)] to the crate attributes to silence this warning
test.rs:5     let _y: u32 = Int::max_value();
                            ^~~~~~~~~~~~~~
```

**Comment by sfackler** [comment]
Nominating

**Comment by eddyb** [comment]
I fixed this during #22172 (and also updated the `run-pass/lint-stability` test) - so I should add "fixes #22641" to the relevant commit.
Aside from, you know, getting the thing merged :disappointed:.

**Comment by eddyb** [comment]
Fixed by #22172.
